### PR TITLE
fix(ValidFieldText): improve validation for numeric input #patch

### PR DESCRIPTION
**Pull request by hamdanhanafi**
this PR was created to fix issue validation inside ValidFieldText, specific inside input type numeric

video attached

https://user-images.githubusercontent.com/94953514/207246866-e0ef1a3d-e1c9-4c48-855d-cb1dbfc8ebd5.mp4

### CODE REVIEW

File: src/components/ValidatedInput/ValidFieldText.tsx [hunk 1]
```diff
@@ -143,8 +145,29 @@ export class ValidFieldText extends React.Component<ValidFieldTextProps, State>
...
+    /**
+     * this section was added for handle validation copy-paste non-numeric characters
+     * onKeydown will skip copy-paste step, so we should add extra validation inside onChange event
+     */
+    if (this.typeIsNumeric() && isNaN(parseInt(target.value))) {
+      // use setTimeout as workaround for fix issue race condition with default behaviour
+      // also we can use Promise, but more simple to use just setTimeout
+      setTimeout(() => {
+        target.value = "";
+      });
```
Comment (rayhan-ab) [inline src/components/ValidatedInput/ValidFieldText.tsx:162]: no timeout as second argument?

Comment (hamdanhanafi) [reply src/components/ValidatedInput/ValidFieldText.tsx:162]: yups no need timeout

**Review by rayhan-ab — APPROVED** [review]

File: src/components/ValidatedInput/ValidFieldText.tsx [hunk 2]
```diff
@@ -143,8 +145,29 @@ export class ValidFieldText extends React.Component<ValidFieldTextProps, State>
...
+  handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const target = event.currentTarget;
+
+    const { onChange } = this.props;
+
+    /**
+     * this section was added for handle validation copy-paste non-numeric characters
+     * onKeydown will skip copy-paste step, so we should add extra validation inside onChange event
+     */
+    if (this.typeIsNumeric() && isNaN(parseInt(target.value))) {
```
Comment (aziz-zen-ab) [inline src/components/ValidatedInput/ValidFieldText.tsx:157]: no check for method `Number.isNaN()` ?

**Review by aziz-zen-ab — APPROVED** [review]

File: src/components/ValidatedInput/ValidFieldText.tsx [hunk 3]
```diff
@@ -192,6 +222,8 @@ export class ValidFieldText extends React.Component<ValidFieldTextProps, State>
     } = this.props;
     const { isFocus } = this.state;
 
+    console.log('propss', this.props)
```
Comment (rayhan-ab) [inline src/components/ValidatedInput/ValidFieldText.tsx:225]: remove console.log

**Pull request merged**